client/src/hooks/graphql/use-with-import-export.tsx

```diff
@@ -0,0 +1,157 @@
+/*
+This software is Copyright ©️ 2020 The University of Southern California. All Rights Reserved. 
+Permission to use, copy, modify, and distribute this software and its documentation for educational, research and non-profit purposes, without fee, and without a written agreement is hereby granted, provided that the above copyright notice and subject to the full license file found in the root of this software deliverable. Permission to make commercial use of this software may be obtained by contacting:  USC Stevens Center for Innovation University of Southern California 1150 S. Olive Street, Suite 2300, Los Angeles, CA 90115, USA Email: accounting@stevens.usc.edu
+
+The full terms of this copyright and license should always be found in the root directory of this software deliverable as "license.txt" and if these terms are not found with this software, please contact the USC Stevens Center for the full license.
+*/
+import { useState } from "react";
+import * as api from "api";
+import {
+  Mentor,
+  MentorExportJson,
+  MentorImportPreview,
+  Question,
+  Subject,
+} from "types";
+import { useActiveMentor } from "store/slices/mentor/useActiveMentor";
+
+export interface UseWithImportExport {
+  mentor: Mentor | undefined;
+  importedJson: MentorExportJson | undefined;
+  importPreview: MentorImportPreview | undefined;
+  exportMentor: () => void;
+  importMentor: (file: File) => void;
+  confirmImport: () => void;
+  cancelImport: () => void;
+  mapSubject: (curSubject: Subject, newSubject: Subject) => void;
+  mapQuestion: (curQuestion: Question, newQuestion: Question) => void;
+}
+
+export function useWithImportExport(accessToken: string): UseWithImportExport {
+  const [importedJson, setImportJson] = useState<MentorExportJson>();
+  const [importPreview, setImportPreview] = useState<MentorImportPreview>();
+  const { mentorState } = useActiveMentor();
+  const mentor = mentorState.data;
+
+  function exportMentor() {
+    if (!mentor) {
+      return;
+    }
+    api.exportMentor(mentor._id).then((m) => {
+      const element = document.createElement("a");
+      element.setAttribute(
```

Comment:
can we avoid the dom manipulation here and do something more `React`-ish? e.g. can this be handled in a use effect? Beyond avoid jquery, I think we should strenuously avoid ever having a hook manipulate the dom as a violation of separation of concerns (hooks manage data, components manage presenation/dom)